pennylane/devices/default_mixed.py

```diff
@@ -796,3 +805,108 @@ def apply(self, operations, rotations=None, **kwargs):
             for k in self.measured_wires:
                 bit_flip = qml.BitFlip(self.readout_err, wires=k)
                 self._apply_operation(bit_flip)
+
+
+@simulator_tracking
+@single_tape_support
+class DefaultMixedNewAPI(Device):
+    r"""A PennyLane Python-based device for mixed-state qubit simulation.
+
+    Args:
+        wires (int, Iterable[Number, str]): Number of wires present on the device, or iterable that
+            contains unique labels for the wires as numbers (i.e., ``[-1, 0, 2]``) or strings
+            (``['ancilla', 'q1', 'q2']``).
+        shots (int, Sequence[int], Sequence[Union[int, Sequence[int]]]): The default number of shots
+            to use in executions involving this device.
+        seed (Union[str, None, int, array_like[int], SeedSequence, BitGenerator, Generator, jax.random.PRNGKey]): A
+            seed-like parameter matching that of ``seed`` for ``numpy.random.default_rng``, or
+            a request to seed from numpy's global random number generator.
+            The default, ``seed="global"`` pulls a seed from NumPy's global generator. ``seed=None``
+            will pull a seed from the OS entropy.
+            If a ``jax.random.PRNGKey`` is passed as the seed, a JAX-specific sampling function using
+            ``jax.random.choice`` and the ``PRNGKey`` will be used for sampling rather than
+            ``numpy.random.default_rng``.
+        r_dtype (numpy.dtype): Real datatype to use for computations. Default is np.float64.
+        c_dtype (numpy.dtype): Complex datatype to use for computations. Default is np.complex128.
+        readout_prob (float): Probability of readout error for qubit measurements. Must be in :math:`[0,1]`.
+    """
+
+    _device_options = ("rng", "prng_key")  # tuple of string names for all the device options.
+
+    @property
+    def name(self):
+        """The name of the device."""
+        return "default.qubit.mixed"
+
+    # pylint: disable=too-many-positional-arguments
+    @debug_logger_init
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        wires=None,
+        shots=None,
+        seed="global",
+        # The following parameters are inherited from DefaultMixed
+        r_dtype=np.float64,
+        c_dtype=np.complex128,
```

Comment:
Not sure if it's worth having these at the moment.  We do have a very backlogged epic to improve float32 support, but I'm not sure that this is worth it for `default.mixed` right now.